javawrapper/device/iothub-client-javawrapper/DeviceClient.java

```diff
@@ -0,0 +1,130 @@
+// Copyright(c) Microsoft.All rights reserved. 
+// Licensed under the MIT license.See LICENSE file in the project root for full license information. 
+
+package javaWrapper;
+
+import com.sun.jna.Pointer;
+
+import javaWrapper.Iothub_client_wrapperLibrary.AMQP_Protocol;
+import javaWrapper.Iothub_client_wrapperLibrary.HTTP_Protocol;
+import javaWrapper.Iothub_client_wrapperLibrary.IOTHUB_CLIENT_HANDLE;
+import javaWrapper.Iothub_client_wrapperLibrary.IOTHUB_CLIENT_RESULT;
+import javaWrapper.Iothub_client_wrapperLibrary.IOTHUB_CLIENT_TRANSPORT_PROVIDER;
+import javaWrapper.Iothub_client_wrapperLibrary.IotHubEventCallback;
+import javaWrapper.Iothub_client_wrapperLibrary.IotHubMessageCallback;
+import javaWrapper.Iothub_client_wrapperLibrary.MQTT_Protocol;
+
+
+public final class DeviceClient 
+{
+	private static IOTHUB_CLIENT_HANDLE handle = new IOTHUB_CLIENT_HANDLE(null);
+	private IOTHUB_CLIENT_TRANSPORT_PROVIDER transport;
+	
+	protected DeviceClient()
+	{
+	
+	}
+	
+	public enum IotHubClientProtocol 
+	{ 
+		HTTPS, AMQPS, MQTT 
```

Comment:
We also have support for AMQP over websockets in C client. We should bring that in as protocol option as well. Including a sample.